### PR TITLE
Week 5 assessment

### PR DESCRIPTION
**Pull request by cmacias56**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ ✅] JavaScript Coding Challenges
- [✅ ] Ruby Coding Challenges
- [ ✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by cmacias56** [comment]
I could not figure out why I was having an issue with .filter on problem 2 on the code challenge

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,39 +16,218 @@
...
+ * spilt the array, .replace the letters with numbers, and .join back together
+ * 
+ * 
+ * ---- conclusion: Ive tried multiple times, rewrote multiple times. Even tried simplifying it. NO-GO :'(
+ */
+
+// const codeMsg = (str) => {       //THIS MAY WORK BUT IM NOT SURE DUE TO THE ERROR IN MY TEST!
+//     let codeStr =
+    
+//     str.replace(/a/g, "4").replace(/e/g, "3").replace(/E/g, "3").replace(/i/g, "1").replace(/o/g, "0")
```
Comment (elysemontano) [inline code-challenges.test.js:57]: You are SUPER close on this and great work for trying out this approach!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,39 +16,218 @@
...
+//  * May need to use a casing method due to Apricot. If I dont, may not pull. toLowerCase
+//  */
+
+const fruits = (arr, letter) => {
+    let fruitArr = arr.filter((str) => {
+    return str.toLowerCase().includes(letter)
+    }) 
+     return fruitArr
+}
+// im not sure why I am getting an error in .filter. I know i need to filter it but not quite sure what im doing wrong. if possible, I would like soe help with this.!!!!!!!!!
```
Comment (elysemontano) [inline code-challenges.test.js:168]: The code looks great, just something to consider in your test.  This particular problem requires two arguments to be passed in, an array and a letter.  Your tests may only be passing one of those...

File: interview-questions.md [hunk 3]
```diff
@@ -8,44 +8,44 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: A gem is a ruby code that is packaged into a component. Each gem has specific functionalities such as built in methods. A gem is literally a GEM for code, they can make a timely process be cut in half to already having the methods needed packed inside
 
 2. What does the acronym CRUD stand for?
 
-Your answer:
+Your answer: The acronym CRUD stands for Create- which allows you to create methods and passes key value pairs! Read - uses methods to find certain information withing a DB.! Update- will store and instance and update it within a variable! Delete- deletes instances in a variable!, CRUD is a great acronym for ruby and rails because it's straight to the point :p! 
 
-Researched answer:
+Researched answer: CRUD stands for create, read, update, delete.
```
Comment (elysemontano) [inline interview-questions.md:19]: Your answer is excellent here, just make sure that you extend your knowledge in the researched answer.

File: ruby-challenges.rb [hunk 4]
```diff
@@ -4,28 +4,113 @@
...
+
+def brake (speed)
+    # @current_speed -= @speed---no-go
+    @current_speed = @current_speed - speed
+if  @current_speed < 0
+    @current_speed = 0
+else @current_speed = @current_speed
+
+end
+end
```
Comment (elysemontano) [inline ruby-challenges.rb:106]: Logic looks good, but just keep an eye on your indentation.

**Review by elysemontano — APPROVED** [review]
Great work on this assessment Christian.  I appreciate you showing the different steps you took in solving each problem.  I left some comments on some of these problems to consider.  In the interview answers, make sure that you are digging deeper into the concepts for the researched answer.  Even if you had the initial answer correct, there is always something you can learn more about that you can write about.

**Pull request merged**